pydicom/tests/test_getdata.py

```diff
@@ -0,0 +1,41 @@
+"""Test the data manager"""
+
+import os
+import unittest
+from os.path import basename
+from pydicom.data import (get_charset_files,
+                          get_testdata_files)
+
+from pydicom.data.base import DATA_ROOT
+
+
+class TestGetData(unittest.TestCase):
+
+    def test_get_dataset(self):
+        """Test the different functions
+        to get lists of data files"""
+
+        # Test base locations
+        charbase = os.path.join(DATA_ROOT, 'charset_files')
+        self.assertTrue(os.path.exists(charbase))
+
+        testbase = os.path.join(DATA_ROOT, 'test_files')
+        self.assertTrue(os.path.exists(testbase))
+
+        # Test file get
+        chardata = get_charset_files()
+        self.assertTrue(len(chardata) > 15)
+
+        # Test that top level file is included
+        bases = [basename(x) for x in chardata]
+        self.assertTrue('charlist.py' in bases)
```

Comment:
The `get_files` function has a filter for a pattern that uses [fnmatch](https://docs.python.org/2/library/fnmatch.html):

```
for root, dirnames, filenames in os.walk(contender):
 +                for filename in fnmatch.filter(filenames, pattern):
 +                    files.append(os.path.join(root, filename))
```

In the current code base, we don't actually use this function as a user would (eg, get me this particular dataset) - we mostly just get specific files under `DATA_ROOT`. I agree that the user of such a function for a particular dataset wouldn't want .py files, so if we extend the data management functions to work to get a dataset for a user, we could add necessary filters. 

Since dicom files are hard to pinpoint in terms of extension, and since we may have data of different extensions / types, I chose this strategy of default grabbing all files, and then allowing for a pattern if needed.